ipaserver/install/installutils.py

```diff
@@ -972,6 +974,71 @@ def check_entropy():
         logger.debug("Invalid value in %s %s", paths.ENTROPY_AVAIL, e)
 
 
+def in_container():
+    """Determine if we're running in a container.
+
+       virt-what will return the underlying machine information so
+       isn't usable here.
+
+       systemd-detect-virt requires the whole systemd subsystem which
+       isn't a reasonable require in a container.
+    """
+    with open('/proc/1/sched', 'r') as sched:
+        data_sched = sched.readline()
```

Comment:
There are envs which mount 'proc' with enabled hidepid option [1]:
> hidepid=n (since Linux 3.3)
      This option controls who can access the information in
      /proc/[pid] directories
    
[1]: https://man7.org/linux/man-pages/man5/proc.5.html

Reply:
Usually I detect this as:
```python
def is_hidepid():
    """Determine if /proc is mounted with hidepid=1/2 option"""
    try:
        os.lstat('/proc/1/stat')
    except (FileNotFoundError, PermissionError):
        return True
    return False
```